addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/internal/core/LxControlColorPickerV2.java

```diff
@@ -0,0 +1,294 @@
+     */
+    public void increaseDecreaseBrightness(int step) {
+        HSBType currentColor = this.getColor();
+        if (currentColor != null) {
+            this.setBrightness(new PercentType(currentColor.getBrightness().intValue() + step));
+        }
+    }
+
+    /**
+     * map the Loxone color to OpenHab HSBType
+     *
+     * @param color color in format hsb(h,s,v) or temp(brightness,kelvin)
+     * @return HSBType
+     */
+    private HSBType mapLoxoneToOH(String color) {
+        HSBType hsbColor = null;
+
+        Matcher m = Pattern.compile("\\((.*?)\\)").matcher(color);
+        if (m.find()) {
+            String[] colorSplit = m.group(1).split(",");
+
+            if (isTemperatureColor(color) && colorSplit.length == 2) {
+                // convert temperature to a OH color
+                int brightness = constrain(Integer.valueOf(colorSplit[0]), 0, 100);
+                int temperature = constrain(Integer.valueOf(colorSplit[1]), 0, 65500);
+
+                int red = map(brightness, 0, 100, 0, calculateRed(temperature));
+                int green = map(brightness, 0, 100, 0, calculateGreen(temperature));
+                int blue = map(brightness, 0, 100, 0, calculateBlue(temperature));
+
+                hsbColor = HSBType.fromRGB(red, green, blue);
+            } else if (isHsvColor(color) && colorSplit.length == 3) {
+                hsbColor = new HSBType(m.group(1));
+            } else {
+                throw new IllegalArgumentException("Invalid color: " + color);
+            }
+        }
+
+        return hsbColor;
+    }
+
+    /**
+     * Re-maps a number from one range to another. That is, a value of fromLow would get mapped to toLow, a value of
+     * fromHigh to toHigh, values in-between to values in-between, etc.
+     *
+     * @param x        the number to map
+     * @param fromLow  the lower bound of the value's current range
+     * @param fromHigh the upper bound of the value's current range
+     * @param toLower  the lower bound of the value's target range
+     * @param toHigh   the upper bound of the value's target range
+     * @return the mapped value
+     */
+    private int map(int x, int fromLow, int fromHigh, int toLow, int toHigh) {
+        return (x - fromLow) * (toHigh - toLow) / (fromHigh - fromLow) + toLow;
+    }
+
+    /**
+     * Constrains a number to be within a range.
+     *
+     * @param x   the number to constrain
+     * @param min the minimum value
+     * @param max the maximum value
+     * @return the constrained value
+     */
+    private int constrain(int x, int min, int max) {
+        if (x >= min && x <= max) {
+            return x;
+        } else if (x < min) {
+            return min;
+        } else {
+            return max;
+        }
+    }
+
+    /**
+     * calculates the red value based on the Kelvin temperature
+     *
+     * @param temp the Kelvin temperature
+     * @return the red value
+     */
+    private int calculateRed(int temp) {
```

Comment:
A minor suggestion: maybe it would be possible to extend the HSBType class constructor with this way of constructing color?